dpcpp/matrix/bccoo_kernels.dp.cpp

```diff
@@ -0,0 +1,205 @@
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************<GINKGO LICENSE>*******************************/
+
+#include "core/matrix/bccoo_kernels.hpp"
+
+
+#include <CL/sycl.hpp>
+
+
+#include <ginkgo/core/base/exception_helpers.hpp>
+#include <ginkgo/core/base/math.hpp>
+#include <ginkgo/core/matrix/csr.hpp>
+#include <ginkgo/core/matrix/dense.hpp>
+
+
+#include "dpcpp/components/format_conversion.dp.hpp"
+
+
+namespace gko {
+namespace kernels {
+/**
+ * @brief DPCPP namespace.
+ *
+ * @ingroup dpcpp
+ */
+namespace dpcpp {
+/**
+ * @brief The Bccoordinate matrix format namespace.
+ *
+ * @ingroup bccoo
+ */
+namespace bccoo {
+
+
+void get_default_block_size(std::shared_ptr<const DefaultExecutor> exec,
+                            size_type *block_size) GKO_NOT_IMPLEMENTED;
+// {
+// 	*block_size = 10;
+// }
+
+
+template <typename ValueType, typename IndexType>
+void spmv(std::shared_ptr<const DpcppExecutor> exec,
+          const matrix::Bccoo<ValueType, IndexType> *a,
+          const matrix::Dense<ValueType> *b,
+          matrix::Dense<ValueType> *c) GKO_NOT_IMPLEMENTED;
+
+GKO_INSTANTIATE_FOR_EACH_VALUE_AND_INDEX_TYPE(GKO_DECLARE_BCCOO_SPMV_KERNEL);
+
+
+template <typename ValueType, typename IndexType>
+void advanced_spmv(std::shared_ptr<const DpcppExecutor> exec,
+                   const matrix::Dense<ValueType> *alpha,
+                   const matrix::Bccoo<ValueType, IndexType> *a,
+                   const matrix::Dense<ValueType> *b,
+                   const matrix::Dense<ValueType> *beta,
+                   matrix::Dense<ValueType> *c) GKO_NOT_IMPLEMENTED;
+
+GKO_INSTANTIATE_FOR_EACH_VALUE_AND_INDEX_TYPE(
+    GKO_DECLARE_BCCOO_ADVANCED_SPMV_KERNEL);
+
+
+template <typename ValueType, typename IndexType>
+void spmv2(std::shared_ptr<const DpcppExecutor> exec,
+           const matrix::Bccoo<ValueType, IndexType> *a,
+           const matrix::Dense<ValueType> *b,
+           matrix::Dense<ValueType> *c) GKO_NOT_IMPLEMENTED;
+
+GKO_INSTANTIATE_FOR_EACH_VALUE_AND_INDEX_TYPE(GKO_DECLARE_BCCOO_SPMV2_KERNEL);
+
+
+template <typename ValueType, typename IndexType>
+void advanced_spmv2(std::shared_ptr<const DpcppExecutor> exec,
+                    const matrix::Dense<ValueType> *alpha,
+                    const matrix::Bccoo<ValueType, IndexType> *a,
+                    const matrix::Dense<ValueType> *b,
+                    matrix::Dense<ValueType> *c) GKO_NOT_IMPLEMENTED;
+
+GKO_INSTANTIATE_FOR_EACH_VALUE_AND_INDEX_TYPE(
+    GKO_DECLARE_BCCOO_ADVANCED_SPMV2_KERNEL);
+
+
+template <typename ValueType, typename IndexType>
+void convert_to_next_precision(
+    std::shared_ptr<const DefaultExecutor> exec,
+    const matrix::Bccoo<ValueType, IndexType> *source,
+    matrix::Bccoo<next_precision<ValueType>, IndexType> *result)
+    GKO_NOT_IMPLEMENTED;
+/*
+{
+
+}
+*/
```

Comment:
Again, remove these comments in this file.

Reply:
Done